indico/htdocs/js/indico/widgets/selectize_widget.js

```diff
@@ -23,26 +23,57 @@
             fieldId: null,
             minTriggerLength: 0,
             searchUrl: null,
-            selectizeOptions: null
+            selectizeOptions: null,
+            allowById: false
         }, options);
 
+        function getSearchData(query) {
+            var m;
+            if (options.allowById && (m = query.match(/^#(\d+)$/))) {
+                return {id: m[1]};
+            } else if (query.length >= options.minTriggerLength) {
+                return {q: query};
+            } else {
+                return null;
+            }
+        }
+
         var params = {
             load: function(query, callback) {
-                if (query.length < options.minTriggerLength) {
+                var data = getSearchData(query);
+                if (!data) {
                     return callback();
                 }
                 $.ajax(options.searchUrl, {
-                    data: {q: query},
+                    data: data,
                     cache: false
                 }).fail(function() {
                     callback();
                 }).done(function(res) {
                     callback(res);
                 });
+            },
+            score: function(query) {
```

Comment:
Otherwise selectize also searches its cache, so if you search for `#10` first and then for `#1` you also see `#10` even though we don't want partial matches when specifying an ID.